enhancements/two-nodes-openshift/2no.md

```diff
@@ -113,32 +119,57 @@ In both cases, the control-plane's dependence on etcd will cause it to respond w
+3. Enabling fencing in RHEL-HA
 
-Otherwise, the procedure follows the standard flow except for the configuration of 2 nodes instead of 3. At this time we've discussed the collection of RedFish details (including passwords!) for each node. This is needed for the RHEL-HA configuration by leveraging the BareMetalHost CRDs populated from the baremetal platform specification in the install-config. There are open questions on how to ensure that pacemaker is the only entity responsible for fencing to prevent conflicting requests to change the machine state between pacemaker and the baremetal operator. Preventing conflicting fencing logic is also important for optional operators like Node Health Check, Self Node Remediation, and Fence Agents Remediation, but these should not be present during installation.
+We propose the inclusion of a new in-cluster operator responsible for the remediation of 2NO resource agents.
 
-An important facility of the installation flow is the transition from a CEO deployed etcd to one controlled by pacemaker. The basic transition works as follows:
+###### Transitioning Etcd Management to RHEL-HA
+An important facility of the installation flow is the transition from a CEO deployed etcd to one controlled by RHEL-HA. The basic transition works as follows:
 1. MCO Extensions are used to ensure that pacemaker, corosync, and resource agents are pre-configured on CoreOS using installation manifests.
-2. Upon detection that the cluster infrastructure is using the DualReplica controlPlaneTopology in the infrastructure config, an in-cluster entity (see open questions regarding whether this should be handled by CEO or an additional operator) will run a command on one of the cluster nodes to initialize pacemaker. The outcome of this is that the resource agent will be started on both nodes.
-3. The aforementioned in-cluster entity will signal CEO to relinquish control of etcd by setting CEO's `managedEtcdKind` to `External`. When this happens, CEO removes the etcd pod from the static pod configs. The resource agents for etcd are running from step 2, and they are configured to wait for etcd pods to be gone so they can restart them using Podman.
+2. Upon detection that the cluster infrastructure is using the DualReplica controlPlaneTopology in the infrastructure config, an in-cluster entity (see open questions regarding whether this should be handled by CEO or the proposed 2NO resource agent operator) will run a command on one of the cluster nodes to initialize pacemaker. The outcome of this is that the resource agent will be started on both nodes.
+3. The aforementioned in-cluster entity will signal CEO to relinquish control of etcd by setting CEO's `managedEtcdKind` to `External`. When this happens, CEO immediately removes the etcd pod from the static pod configs. The resource agents for etcd are running from step 2, and they are configured to wait for etcd pods to be gone so they can restart them using Podman.
```

Comment:
What happens if the etcd pod definition need to change after etcd is started using Podman, who is in charge for updating and restarting it?

Reply:
the same holds also for certificates - I would leave this for later (GA).

Reply:
Do we still need etcd to render the static pod definition, but not the the static pod directory?

Reply:
I've updated this section to with some of the background from this thread.

Reply:
> Do we still need etcd to render the static pod definition, but not the the static pod directory?

I don't understand the question, can you clarify?

Reply:
My idea is that since CEO would normally roll out updates to etcd, it would normally push those updates out as changes to the static pod definition. Instead of doing that, what we might be able to do is have etcd still render that the block related to the etcd pod, except that it would write it to disk somewhere other than the static pod location. Then, all pacemaker has to do is monitor changes to the CEO-updated file and redeploy whenever that file is touched.

Reply:
if the update refers to image, we can make the agent able to pick it automatically (e.g. looking into the image definition of `etcdctl` container, which uses the same image as `etcd`) at each startup, so the image would be always up to date.
For different changes, we can define categories that the agent will be able to "understand" (for example I'm thinking about the value of some environment variables it uses)